File: 6_Sum_of_square_difference/python/main.py
def sum_square_upto(n: int) -> int:
    return sum(map(lambda x: x**2, range(1, n+1)))

def square_sum_upto(n: int) -> int:
    return sum(range(1, n+1))**2

def main():
    n = 100
    print(square_sum_upto(n) - sum_square_upto(n))

if __name__ == "__main__":
    main()

File: 6_Sum_of_square_difference/rust/src/main.rs
fn main() {
    let n: i32 = 100;
    println!("{:?}", square_sum_upto(n) - sum_square_upto(n));
}

fn sum_square_upto(n: i32) -> i32 {
    (1..n + 1)
        .into_iter()
        .map(|x| x.pow(2))
        .collect::<Vec<i32>>()
        .iter()
        .sum()
}

fn square_sum_upto(n: i32) -> i32 {
    (1..n + 1).into_iter().sum::<i32>().pow(2)
}
